src/logic/schedule-exporter/workers-absence-export.logic.ts

```diff
@@ -0,0 +1,162 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import xlsx from "exceljs";
+import { ScheduleDataModel } from "../../state/schedule-data/schedule-data.model";
+import { EMPTY_ROW, ABSENCE_HEADERS } from "../../helpers/parser.helper";
+import { CELL_MARGIN } from "./schedule-export.logic";
+import { ShiftCode, SHIFTS } from "../../state/schedule-data/shifts-types/shift-types.model";
+import { TranslationHelper } from "../../helpers/translations.helper";
+import {
+  DateInformationForWorkInfoCalculation,
+  WorkerHourInfo,
+} from "../schedule-logic/worker-hours-info.logic";
+import { PrimaryMonthRevisionDataModel } from "../../state/application-state.model";
+import { MonthInfoLogic } from "../schedule-logic/month-info.logic";
+import { WeekDay } from "../../state/schedule-data/foundation-info/foundation-info.model";
+
+export class WorkersAbsenceExportLogic {
+  private scheduleModel: ScheduleDataModel;
+
+  private revision: PrimaryMonthRevisionDataModel;
+
+  private cellsToMerge: number[][];
+
+  constructor(scheduleModel: ScheduleDataModel, revision: PrimaryMonthRevisionDataModel) {
+    this.scheduleModel = scheduleModel;
+    this.revision = revision;
+    this.cellsToMerge = [];
+  }
+
+  public setWorkersWorkSheet(workSheet: xlsx.Worksheet): void {
+    workSheet.pageSetup.showGridLines = true;
+    workSheet.pageSetup.fitToPage = true;
+    workSheet.pageSetup.fitToHeight = 1;
+    workSheet.pageSetup.fitToWidth = 1;
+    workSheet.pageSetup.horizontalCentered = true;
+
+    const workersInfoArray = this.createWorkersInfoSection(this.scheduleModel, this.revision);
+
+    const colLens = workersInfoArray[0].map((_, colIndex) =>
+      Math.max(...workersInfoArray.map((row) => row[colIndex]?.toString().length ?? 0))
+    );
+
+    workSheet.addRows(workersInfoArray);
+
+    this.cellsToMerge.forEach((pair, index) => {
```

Comment:
Podoba mi się mergowanie dla kolumny pracownika, ale niestety nie może mergować godzin i roku. 
Widzę tutaj 2 problemy: 
1. Urlop może być za inny rok w każdym przypadku 
2. Z tego co widzę to nie liczą godzin dla zwolenień lekarskich, izolacji itd. To raczej też nie jest liczone za jakiś rok. Nie wiem z czego to wynika do końca. Musimy się dowiedzieć.